include/fix8/sessionwrapper.hpp

```diff
@@ -248,10 +248,16 @@ class ClientSession : public ClientSessionBase
 	/// Dtor.
 	virtual ~ClientSession ()
 	{
-		delete _persist;
+		session_ptr()->clear_connection( _cc );
 		delete _session;
-		delete _log;
+		_session = nullptr;
+		_persist->stop();
```

Comment:
The ClientSession object is not designed to be reused. It should be instantiated for each Session, even on reconnect, unless you are using ReliableClientSession which does reuse Session but manages that itself.